### PR TITLE
Update robots.txt.dist

### PR DESCRIPTION
**Pull request by Kostelano**
Not index links to restore the login, password, etc:

http://SITE.COM/component/users/?view=login
http://SITE.COM/component/users/?view=registration
http://SITE.COM/component/users/?view=remind
http://SITE.COM/component/users/?view=reset

Send the article to a friend
http://SITE.COM/component/mailto/?tmpl=component&template=protostar&link=6ea8583f082e256dc6b945104b150650da776e58

---

Not index content that is linked to categories "uncategorised". This page is a duplicate.

---

Not index links:

http://SITE.COM/YOUR_CATEGORY?limitstart=0
http://SITE.COM/YOUR_CATEGORY?start=5
http://SITE.COM/YOUR_CATEGORY?start=10

---

Irrelevant information on the page http://tool.motoricerca.info/robots-checker.phtml, symbol \* can be used.
http://savepic.su/5070990.png

### CODE REVIEW

**Comment by brianteeman** [comment]
> Not index content that is linked to categories "uncategorised". This page is a duplicate.

How is this automatically duplicate content?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6098">issues.joomla.org/joomla-cms/6098</a>.</sub>

File: robots.txt.dist [hunk 1]
```diff
@@ -26,4 +24,6 @@ Disallow: /logs/
 Disallow: /modules/
 Disallow: /plugins/
 Disallow: /tmp/
-
++Disallow: /*print*
```
Comment (zero-24) [inline robots.txt.dist]: @Kostelano are you sure to add a `+` here?

File: robots.txt.dist [hunk 2]
```diff
@@ -7,15 +7,13 @@
...
-#
-# For syntax checking, see:
-# http://tool.motoricerca.info/robots-checker.phtml
 
 User-agent: *
 Disallow: /administrator/
 Disallow: /bin/
 Disallow: /cache/
 Disallow: /cli/
+Disallow: /component/
```
Comment (mbabker) [inline robots.txt.dist]: Not sure I would do this.  It effectively blocks indexing for all SEF routes that aren't associated with the menu system (so if sites are using routes like `/component/content/article/42` then their content won't get indexed)

File: robots.txt.dist [hunk 3]
```diff
@@ -26,4 +24,6 @@ Disallow: /logs/
 Disallow: /modules/
 Disallow: /plugins/
 Disallow: /tmp/
-
++Disallow: /*print*
++Disallow: /*start*
++Disallow: /*uncategorised*
```
Comment (mbabker) [inline robots.txt.dist]: Also not sure I would add this one.  Perhaps users are intending to publish content to an uncategorised category and want the content to be picked up via an indexer.

Comment (Bakual) [inline robots.txt.dist]: On a related note, why disallow *start* and *print*? There may well be menu items with those terms in it.

**Comment by bertmert** [comment]
All these new lines are NOT robots.txt standard.

http://www.robotstxt.org/robotstxt.html
"Note also that globbing and regular expression are not supported in either the User-agent or Disallow lines. The '*' in the User-agent field is a special value meaning "any robot". Specifically, you cannot have lines like "User-agent: *bot*", "Disallow: /tmp/*" or "Disallow: *.gif". "

And I've never seen a plus sign in robots.txt standards. What is it good for?

**Comment by Bakual** [comment]
@Kostelano Can you elaborate and answer the raised question?
To me it doesn't make much sense what you propose here.

File: robots.txt.dist (hunk 2 above)
Comment (Fedik) [inline robots.txt.dist]: as I understand it is the main purpose of this PR,
if you have article linked to menu, then on the site you have 2 valid link, eg duplication:
`/blog/42-blabla` and `/component/content/article/42`

Comment (Bakual) [inline robots.txt.dist]: Just for reference: Google has no issues per se with multiple URLs for the same content. As long as there is a proper canonical tag and it doesn't look like you do it on purpose to get better ranking.

**Comment by Kostelano** [comment]
* About the "uncategorised".
Almost always the category "Uncategorized" is not renamed. Any articles that are tied to this category, and thus get a link from the menu (for example) have a double:

http://SITE.COM/content_1/
http://SITE.COM/content_1/2-uncategorised/

This almost always takes place.

* Sorry, copying accidentally added "+" symbol. Of course, it's not there! I'm sorry!

After listening to your opinions, I suggest to specify some strings so that it does not create problems with the menus (for example) - someone wrote above.
For example:

    Disallow: /*?tmpl=component&print=*
    Disallow: /*?start=*
    Disallow: /*?limitstart=* // it duplicates the first page category with page navigation
    Disallow: /*-uncategorised

Can be ignored "?start=" because the line is not a double. This page is just not interesting to users.

* Perhaps google and well takes two identical pages (for Bakual), but why do we? It is logical to have one such page is correct. With that, adding the exception string with /component/, we get rid of at least five more unnecessary pages in the index of search engines.

After all, even with the authorization module off the search engine spiders find links to the registration, login and password recovery, etc.

Thank you for your attention.

**Comment by mbabker** [comment]
> Perhaps google and well takes two identical pages (for Bakual), but why do we? It is logical to have one such page is correct. With that, adding the exception string with /component/, we get rid of at least five more unnecessary pages in the index of search engines.

Like I noted, your change with that line would effectively block search engines from not indexing ALL routes using the default Joomla SEF implementation (where routes are named `/component/<component_name>/<view_name>`) and I don't think that is a change that we should make to our default robots.txt file.  For what you're suggesting to accomplish, I really think you need more specific rules that aren't going to cause potential issues with other valid routes.

**Comment by richard67** [comment]
As stated above by @bertmert (but ignored during further dicsussions), the globbing in the disallow lines is NOT robots.txt standard. It is a Google extension which is not standard and not supported by most other search engines.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6098">issues.joomla.org/joomla-cms/6098</a>.</sub>

**Comment by Bakual** [comment]
Based on the overall feedback so far, I'm closing this PR.

Feel free to create a new one with the feedback included.

**Pull request closed without merging**